site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -0,0 +1,217 @@
+                else{
+                    $this->core->addErrorMessage("Unable to add to queue");
+                }
+
+            }
+            else{
+                $this->core->addErrorMessage("You are already in the queue");
+            }
+        }else{
+            if($_POST['name'] == ""){
+                $this->core->addErrorMessage("Invalid Name");
+            }
+            else if(is_null($section_id)){
+                $this->core->addErrorMessage("Invalid Code");
+            }
+            else if(!$this->core->getQueries()->isQueueOpen()){
+                $this->core->addErrorMessage("Queue is closed");
+            }
+
+        }
+        return Response::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+        );
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/startHelp", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
+    * @return Response
+    */
+    public function startHelpPerson(){
+        if(!isset($_POST['entry_id'])){
+            $this->core->addErrorMessage("Missing entry ID");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if($this->core->getUser()->accessGrading())
+        $this->core->getQueries()->startHelpUser($_POST['entry_id']);
+        return Response::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+        );
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/finishHelp", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
+    * @return Response
+    */
+    public function finishHelpPerson(){
+        if(!isset($_POST['entry_id'])){
+            $this->core->addErrorMessage("Missing entry ID");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if($this->core->getUser()->accessGrading())
+        $this->core->getQueries()->finishHelpUser($_POST['entry_id'], $this->core->getUser()->getId());
+        return Response::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+        );
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/toggle", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
+    * @return Response
+    */
+    public function toggleQueue(){
+        if(!isset($_POST['queue_open'])){
+            $this->core->addErrorMessage("Missing queue status");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if(!$this->core->getUser()->accessGrading()){
```

Comment:
You should be able to get rid of this if statement check as that should be self-contained within the `@AccessControl` annotation test.